packages/compiler/src/server/compile-service.ts

```diff
@@ -174,12 +177,60 @@ export function createCompileService({
       if (serverHost.throwInternalErrors) {
         throw err;
       }
+
+      let uri = document.uri;
+      let range = Range.create(0, 0, 0, 0);
+      if (err.info.kind === "emitter" && configFilePath) {
+        const sourceFile = await serverHost.compilerHost.readFile(configFilePath);
+        const emitterName = err.info.metadata.name;
+        const yamlDoc = parseDocument(sourceFile.text, {
+          keepSourceTokens: true,
+        });
+        if (yamlDoc) {
+          let emitOffset = 0;
```

Comment:
it looks more complicated than i thought. Doesn't yamlDoc provide way to access its element?